stats.go

```diff
@@ -1,23 +1,43 @@
 package main
 
+import "sync/atomic"
+
 type Stats struct {
-	ServerStats
+	ServerStats `json:"server_stats"`
+}
+
+func (s *Stats) Copy() *Stats {
+	newS := &Stats{
+		ServerStats{
+			atomic.LoadUint64(&s.TotalURL),
+			Redirects{
+				atomic.LoadUint64(&s.Redirects.Success),
+				atomic.LoadUint64(&s.Redirects.Failed),
+			},
+			make([]Handler, len(s.Handlers)),
+		},
+	}
+
+	for i, v := range s.Handlers {
+		newS.Handlers[i] = Handler{v.Name, atomic.LoadUint64(&v.Count)}
+	}
```

Comment:
here you can use the `copy` builtin to copy the slice

Reply:
If I copy the slice with the `copy` primitive the counter will be copied atomically? If not it maybe lead to data-race. However, I think that in the x86 platform a words will be always read atomically.

Reply:
So all in all, the use of atomic is required here for correctness.
Here a link that might be of interest about that:
https://dave.cheney.net/2018/01/06/if-aligned-memory-writes-are-atomic-why-do-we-need-the-sync-atomic-package

Also go has a race detector, which comes very handy. `go run -race` or `go build -race`